scripts/tests/test_validate_unwanted_patterns.py

```diff
@@ -0,0 +1,77 @@
+import io
+
+# the "as" is just so I won't have to change this in many places later.
+import validate_string_concatenation as validate_unwanted_patterns
+
+
+class GoodBarePytestRaises:
+    def normal_pytest_raises(self):
+        return io.StringIO(
+            """
+with pytest.raises(ValueError, match="foo"):
+    pass
+"""
+        )
+
+    def pytest_raises_as_comment(self):
+        return io.StringIO(
+            """
+# with pytest.raises(ValueError, match="foo"):
+#     pass
+"""
+        )
+
+    def bare_pytest_raises_as_comment(self):
+        """
+        We do not care if the bare pytest raises is commented out.
+        """
+        return io.StringIO(
+            """
+# with pytest.raises(ValueError):
+#     pass
+"""
+        )
+
+
+class BadBarePytestRaises:
+    def normal_bare_pytest_raises(self):
+        return io.StringIO(
+            """
+with pytest.raises(ValueError):
+    raise ValueError("foo")
+"""
+        )
+
+
+class TestGoodBarePytestRaises:
+    def test_good_normal_pytest_raises(self):
+        result = validate_unwanted_patterns.bare_pytest_raises(
+            GoodBarePytestRaises.normal_pytest_raises(self)
```

Comment:
Sorry the tests I told you to use as reference were too complicated. This is actually easier. Something like this is probably what we want:
```python
def test_bare_raises():
    fd = io.StringIO("""with pytest.raises(ValueError):
        raise ValueError("foo")""")
    result = list(validate_unwanted_patterns.bare_pytest_raises(fd))
    expected = [(1, "Bare pytests raise have been found. Please pass in the argument 'match' as well the exception.")]
    assert result == expected
```
The idea is that when our test suite executes, it'll call this function. This function will call the one you implemented with a known input, and this function will raise an assertion error, if your function is not returning what it's expected to return.

It's also great documentation to understand what your function is expected to receive as an input, and how the output looks like.

Reply:
@datapythonista Is it a good idea to not have the error message hard coded into the tests, but instead defining constants for each error message, in ```scripts/validate_unwanted_patterns```. ?

So if for some reason we want to change the error message we won't have to change it in many placed, and only in one place.

Reply:
I see your point, but I don't think that's common, and I wouldn't do that.

Having repetition in the tests is good. If you are reporting the wrong message in one exception, and you test that you're using the same constant you report, your test will pass, and the message will be incorrect. Better to keep things simple and explicit.

Reply:
Thank you for the explanation.